### PR TITLE
fix painting compile error

### PR DESCRIPTION
**Pull request by SiyaoHuang**

### CODE REVIEW

File: com.unity.uiwidgets/Runtime/painting/text_painter.cs [hunk 1]
```diff
@@ -29,21 +63,36 @@ public class TextPainter {
...
             _textScaleFactor = textScaleFactor;
             _maxLines = maxLines;
             _ellipsis = ellipsis;
             _strutStyle = strutStyle;
+            _textWidthBasis = textWidthBasis;
+            _textHeightBehavior = textHeightBehavior;
+        }
+
+        void markNeedsLayout() {
+            _paragraph = null;
```
Comment (zhuxingwei) [inline com.unity.uiwidgets/Runtime/painting/text_painter.cs:92]: _paragraph is a C# object which will hold a c++ object. I see no place to call _paragraph._dispose() in this class, is there any memory leak issue on the c++ side here?

Comment (SiyaoHuang) [reply com.unity.uiwidgets/Runtime/painting/text_painter.cs:92]: Paragraph implements NativeWrapper, deconstructor of which calls _dispose.

File: com.unity.uiwidgets/Runtime/painting/gradient.cs [hunk 2]
```diff
@@ -152,15 +153,15 @@ public class LinearGradient : Gradient, IEquatable<LinearGradient> {
 
         public readonly TileMode tileMode;
 
-        public override PaintShader createShader(Rect rect) {
+        public override Shader createShader(Rect rect, TextDirection? textDirection = null) {
```
[the remark this answers is not in the document]
Comment (SiyaoHuang) [reply com.unity.uiwidgets/Runtime/painting/gradient.cs:192]: updated createShaders. AlignmentGeometry is not fully implemented.

File: com.unity.uiwidgets/Runtime/painting/image_provider.cs [hunk 3]
```diff
@@ -167,10 +176,11 @@ public abstract class ImageProvider<T> : ImageProvider {
             ImageStream stream = new ImageStream();
             T obtainedKey = default;
 
-            obtainKey(configuration).Then(key => {
+            obtainKey(configuration).then_((T key) => {
                 obtainedKey = key;
-                stream.setCompleter(PaintingBinding.instance.imageCache.putIfAbsent(key, () => load(key)));
-            }).Catch(ex => {
+                // TODO : how to load
```
[the remark this answers is not in the document]
Comment (SiyaoHuang) [reply com.unity.uiwidgets/Runtime/painting/image_provider.cs:181]: Updated

**Review by zhuxingwei — APPROVED** [review]

**Pull request merged**